feature_tests/steps/result_steps.py

```diff
@@ -0,0 +1,87 @@
+from behave import then
+from domain.core.reference import Reference
+
+from feature_tests.steps.common import parse_value, read_value_from_path
+
+
+@then("the result is of type {type_name}")
+def step_impl(context, type_name):
+    t = type(context.result).__name__
+    assert t == type_name, f"Expected '{type_name}' got '{t}'"
+
+
+@then("the result {field} equals {value:Number}")
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
```

Comment:
more generally the following is preferred, since __dict__ doesn't always return attributes:

```suggestion
    v = getattr(result, field)
```